OpenImis.ModulesV2/InsureeModule/Repositories/FamilyRepository.cs

```diff
@@ -0,0 +1,273 @@
+                                        && D.ValidityTo == null)
+                                    select F.FamilyId)
+                                   .FirstOrDefault();
+
+                    response = imisContext.TblInsuree
+                                        .Where(i => i.ValidityTo == null && i.FamilyId == familyId)
+                                        .Join(imisContext.TblFamilies, i => i.FamilyId, f => f.FamilyId, (i, f) => f)
+                                        .Where(f => f.ValidityTo == null)
+                                        .Include(f => f.TblInsuree)
+                                            .ThenInclude(f => f.Photo)
+                                        .Select(f => FamilyModel.FromTblFamilies(f))
+                                        .FirstOrDefault();
+                }
+
+                if (response == null)
+                {
+                    return null;
+                }
+
+                return response;
+            }
+            catch (SqlException e)
+            {
+                throw e;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        public int Create(EnrollFamilyModel model, int userId, int officerId)
+        {
+            try
+            {
+                string webRootPath = _hostingEnvironment.WebRootPath;
+
+                var XML = model.GetEnrolmentFromModel().Serialize();
+
+                var EnrollmentDir = _configuration["AppSettings:Enrollment_Phone"];
+                var JsonDebugFolder = _configuration["AppSettings:JsonDebugFolder"];
+                var UpdatedFolder = _configuration["AppSettings:UpdatedFolder"];
+                var SubmittedFolder = _configuration["AppSettings:SubmittedFolder"];
+
+                var hof = "";
+
+                if (model.Insuree.Any(x => x.isHead == "true" || x.isHead == "1"))
+                {
+                    hof = model.Insuree.Where(x => x.isHead == "true" || x.isHead == "1").Select(z => z.CHFID).FirstOrDefault();
+                }
+                else hof = "Unknown";
+
+                var FileName = string.Format("{0}_{1}_{2}.xml", hof, officerId.ToString(), DateTime.Now.ToString("dd-MM-yyyy HH-mm-ss.XML"));
+                var JsonFileName = string.Format("{0}_{1}_{2}.xml", hof, officerId.ToString(), DateTime.Now.ToString("dd-MM-yyyy HH-mm-ss"));
+
+
+                var xmldoc = new XmlDocument();
+                xmldoc.InnerXml = XML;
+
+                try
+                {
+                    if (!Directory.Exists(webRootPath + EnrollmentDir)) Directory.CreateDirectory(webRootPath + EnrollmentDir);
+
+                    xmldoc.Save(webRootPath + EnrollmentDir + FileName);
+
+                    if (!Directory.Exists(webRootPath + JsonDebugFolder)) Directory.CreateDirectory(webRootPath + JsonDebugFolder);
+
+                    File.WriteAllText(webRootPath + JsonDebugFolder + JsonFileName, XML);
```

Comment:
JSON file should have JSON content, not XML.

Reply:
Done